### ISSUE TITLE
`prefer_asserts_in_initializer_lists` lint is triggered even if there is non-assertion code in the constructor body

### ISSUE THREAD

**Issue by elliette** [dart-lang/linter#3156]
For the following example constructor (below), the `prefer_asserts_in_initializer_lists` is being triggered, though when I look at the [lint definition](https://dart-lang.github.io/linter/lints/prefer_asserts_in_initializer_lists.html) it says: "DO put asserts in initializer list for constructors **with only asserts in their body**." 

Since the following constructor body has more than just the `assert`, I would assume that the lint wouldn't be triggered. 

```
  FileSearchResults.withQuery({
    @required this.query,
    @required this.allScripts,
  }) {
    assert(!query.isEmpty);

    for (final scriptRef in allScripts) {
      if (query.isExactFullPathMatch(scriptRef)) {
        _exactFullPathMatches.add(scriptRef);
      } else if (query.isFuzzyMatch(scriptRef)) {
        _fuzzyMatches.add(scriptRef);
      }
    }
  }
```

**Comment by pq** [comment]
It's interesting.  Looking at the source it appears that this is WAI but that doesn't square w/ the docs.  Needless to say, something isn't right!

FWIW, here's where the lint was introduced: https://github.com/dart-lang/linter/pull/677.

@a14n: do you have any thoughts?

Thanks for the report @elliette!

**Comment by a14n** [comment]
I guess the documentation should be updated. The lint rule [evolved](https://github.com/dart-lang/linter/pulls?q=is%3Apr+prefer_asserts_in_initializer_list+is%3Aclosed) but not its documentation. "DO put asserts in initializer list." should be more correct.

**Comment by pq** [comment]
Thanks @a14n!
